### PR TITLE
Added a route to serve the swagger API definition as YAML.

### PR DESCRIPTION
**Pull request by cjllanwarne**
Added the YAML swagger spec.
Moved some classes around and standardised some of the naming w.r.t Cromwell.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -13,7 +13,6 @@ val downgradedSprayV = "1.3.1"
 val akkaV = "2.3.12"
 
 libraryDependencies ++= Seq(
-  "com.gettyimages" %% "spray-swagger" % "0.5.1",
   "org.webjars" % "swagger-ui" % "2.1.1",
```
Comment (cjllanwarne) [inline build.sbt]: Am I right that we don't want the swagger-ui here either?

File: src/main/resources/yaml/thurloe-api.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
+swagger: "2.0"
+info:
+  title: Thurloe Service
+  version: 0.1
+paths:
+  /thurloe/{userId}:
+    get:
+      operationId: getAll
+      summary: Returns a list of all keys and values stored in Thurloe.
```
Comment (mcovarr) [inline src/main/resources/yaml/thurloe-api.yaml]: for a specified user id

File: src/main/resources/yaml/thurloe-api.yaml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        200:
+          description: OK
+        400:
+          description: Bad request (e.g. invalid JSON input)
+        500:
+          description: Internal server error
+  /thurloe/{userId}/{key}:
+    get:
+      operationId: getKvp
+      summary: Returns a single key/value pair, looked up by key.
```
Comment (mcovarr) [inline src/main/resources/yaml/thurloe-api.yaml]: and user id

File: src/main/resources/yaml/thurloe-api.yaml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      responses:
+        200:
+          description: OK
+        404:
+          description: Key not found
+        500:
+          description: Internal server error
+    delete:
+      operationId: deleteKvp
+      summary: Deletes a single key/value pair, referenced by key.
```
Comment (mcovarr) [inline src/main/resources/yaml/thurloe-api.yaml]: and user id

File: src/main/resources/yaml/thurloe-api.yaml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      produces:
+      - text/plain
+      parameters:
+      - name: userId
+        in: path
+        description: The user ID of the requester
```
Comment (mcovarr) [inline src/main/resources/yaml/thurloe-api.yaml]: the user ID whose key is being requested, not sure this is necessarily the requester

File: src/main/resources/yaml/thurloe-api.yaml [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        description: The key of the data to return
+        required: true
+        type: string
+      responses:
+        200:
+          description: OK
+        404:
+          description: Key not found
+        500:
+          description: Internal server error
```
Comment (mcovarr) [inline src/main/resources/yaml/thurloe-api.yaml]: Is there a way of specifying that the returned format for the happy paths looks like `KeyValuePair`?  There was a syntax for doing this with the annotations but it sometimes caused compile problems so we didn't always use it.  :rage:

Comment (cjllanwarne) [inline src/main/resources/yaml/thurloe-api.yaml]: You probably didn't import enough implicits!

Comment (mcovarr) [inline src/main/resources/yaml/thurloe-api.yaml]: True, you can never have too many implicits.

Comment (cjllanwarne) [inline src/main/resources/yaml/thurloe-api.yaml]: I've managed this for the input for the POST... I'll see if the same format works for the returns for GET

Comment (mcovarr) [inline src/main/resources/yaml/thurloe-api.yaml]: nice!

File: build.sbt (hunk 1 above)
Comment (mcovarr) [inline build.sbt]: Long term with a centralized Swagger it should go, short term we might need to hang onto it a bit longer

File: src/main/resources/yaml/thurloe-api.yaml (hunk 6 above)
Comment (cjllanwarne) [inline src/main/resources/yaml/thurloe-api.yaml]: Done! See other routes for the JSON schemas (DELETE doesn't return any JSON).

Comment (mcovarr) [inline src/main/resources/yaml/thurloe-api.yaml]: Great, and I'm so glad this is no longer in code.  :smile:

**Pull request merged**